src/reducers/calculatorReducer.js

```diff
@@ -0,0 +1,131 @@
+import { CLEAR, EVALUATED, DECIMAL_INPUT, NUMBER_INPUT, OPERATOR_INPUT } from "../types";
+
+const initialState = {
+    currentVal: "0",
+    prevVal: "0",
+    formula: "",
+    evaluated: false
+}
+
+const isOperator = /[*/+-]/,
+    endWithOperator = /([*/+-])$/,
+    endWithNegative = /([*/+])-$/,
+    endWithDigitAndNegative = /(\d+[-])$/
+
+const calculatorReducer = (state = initialState, action) => {
+    switch (action.type) {
+        case NUMBER_INPUT:
+            if (state.evaluated) {
+                return {
+                    ...state,
+                    currentVal: action.number,
+                    formula: action.number === "0" ? "" : action.number,
+                    evaluated: false
+                }
+            }
+            else {
+                return {
+                    ...state,
+                    currentVal: state.currentVal === "0" || isOperator.test(state.currentVal) ? action.number : state.currentVal + action.number,
+                    formula: state.currentVal === "0" && action.number === "0" ? action.number : state.formula + action.number
+                }
```

Comment:
please try to have one return. make the condition and ternary operator is simpler